src/Interpreters/Session.cpp

```diff
@@ -401,13 +406,20 @@ ContextMutablePtr Session::makeSessionContext()
     prepared_client_info.reset();
 
     /// Set user information for the new context: current profiles, roles, access rights.
+    /// TCP handler quota is received after context creation.
+    /// Without quota key new_session_context->setUser(*user_id, true) can throw exception
```

Comment:
The comment on l. 410 might confuse future readers, maybe remove it?

Reply:
Done